### PR TITLE
Use elasticsearch for device log users filter

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Previously, the users filter select2 was populated by querying unique values in the username column of the `DeviceReportEntry` table. However, this table can have 10s of millions of rows, so the query is slow. This changes that select2 to be populated by elasticsearch instead.

@dannyroberts 
http://manage.dimagi.com/default.asp?254116

### CODE REVIEW

File: corehq/apps/reports/filters/api.py [hunk 1]
```diff
@@ -271,17 +271,37 @@ class DeviceLogFilter(LoginAndDomainMixin, JSONResponseMixin, View):
...
+        query_set = (
             DeviceReportEntry.objects
             .filter(domain=domain)
             .filter(**field_filter)
             .distinct(self.field)
             .values_list(self.field, flat=True)
-        )[:10]
+            .order_by(self.field)
+        )
+        values = query_set[self._offset():self._offset()+self._page_limit()]
```
Comment (NoahCarnahan) [inline corehq/apps/reports/filters/api.py]: The endpoint always returned just the first page before

File: corehq/apps/reports/filters/api.py [hunk 2]
```diff
@@ -271,17 +271,37 @@ class DeviceLogFilter(LoginAndDomainMixin, JSONResponseMixin, View):
...
         return self.render_json_response({
             'results': [{'id': v, 'text': v} for v in values],
+            'total': query_set.count(),
```
Comment (NoahCarnahan) [inline corehq/apps/reports/filters/api.py]: This is a required field for the select2 ajax functionality

File: corehq/apps/reports/filters/api.py [hunk 3]
```diff
@@ -271,20 +272,53 @@ class DeviceLogFilter(LoginAndDomainMixin, JSONResponseMixin, View):
...
+        query_set = (
             DeviceReportEntry.objects
             .filter(domain=domain)
             .filter(**field_filter)
             .distinct(self.field)
             .values_list(self.field, flat=True)
-        )[:10]
+            .order_by(self.field)
+        )
+        values = query_set[self._offset():self._offset()+self._page_limit()]
```
Comment (dimagimon) [inline corehq/apps/reports/filters/api.py]: E226 missing whitespace around arithmetic operator

**Review by dannyroberts — APPROVED** [review]
neat

**Comment by snopoke** [comment]
:candy:

**Pull request merged**